soql-environment/src/main/scala/com/socrata/soql/environment/Source.scala

```diff
@@ -0,0 +1,58 @@
+package com.socrata.soql.environment
+
+import scala.util.parsing.input.{Position, NoPosition}
+
+import com.rojoma.json.v3.codec.{JsonEncode, JsonDecode}
+import com.rojoma.json.v3.util.{SimpleHierarchyEncodeBuilder, SimpleHierarchyDecodeBuilder, InternalTag, AutomaticJsonEncodeBuilder, AutomaticJsonDecodeBuilder}
+
+import com.socrata.soql.jsonutils.HierarchyImplicits._
+import com.socrata.soql.jsonutils.PositionCodec
+
+sealed abstract class Source[+RNS] {
+  val scopedResourceName: Option[ScopedResourceName[RNS]]
```

Comment:
Is `RNS` short for `ResourceNameScope`?

Reply:
Yep; it's an abbreviation used pervasively in the code.